# Example 8.11
A car is linear moving with constant acceleration $a\,\mathrm{m/s^2}$. Let $s_0$ denote the car's distance to a reference point 0 at the time $t=0$,and $v_0$ denotes its speed at this time. The function $s(t)$ gives the distance from reference point at time t. The function $s(t)$ can be given though $s(t) = s_0 + v_0 t + at^2/2$.

<b>a)</b> Calculate the unknown parameters for the following measurements.

In [1]:
from math import *
import numpy as np

In [2]:
t = (2.,5.,9.)
s = (8.7, 40.9, 125.7)

The unknown $s_0, v_0, a$ can be calculated from a linear system using the three measurements. We setup a coefficient matrix and solve the linear system.

In [3]:
A = np.array([[1.,t[0],0.5*t[0]**2],
              [1.,t[1],0.5*t[1]**2],
              [1.,t[2],0.5*t[2]**2]])
x = np.linalg.solve(A,s)
s_0 = x[0]; v_0 = x[1]; a = x[2]
's_0=%.4f m  v_0=%.4f m/s  a=%.4f m/s**2' % (s_0, v_0, a)

's_0=2.1857 m  v_0=0.2667 m/s  a=2.9905 m/s**2'

<b>b)</b> Estimate the error of the parameters for an error in s values of at most $\pm 0.05$.

The condition number specifies the error propagation $$K=\frac{R_{out}}{R_{in}}.$$

In [4]:
K = 202
d_s = 0.5E-1
R_in = d_s/np.linalg.norm(s)
R_out = K*R_in
print "R_out = %.2f" %R_out

R_out = 0.08


In [5]:
E_out = np.linalg.norm(x)*R_out
"E_out = %.2f"  % E_out

'E_out = 0.28'

The maximal error in the unknown parameter is $E_{\mathrm{out}}=0.28$.

<b>c)</b> Estimation of the total error with addtional errors in t-values of at most $\pm 0.01$. Use experimental error estimation.